File: distributions/1D_beta50020_input.R
# ============================================================
# 1D_beta50020_input.R
# Beta(500, 20)
# ============================================================

extract_samples <- function(n) {
  return(matrix(rbeta(n, 500, 20), ncol = 1))
}

compute_pdf <- function(x) {
  dbeta(x, 500, 20)
}

File: distributions/1D_beta50020_input.py
import numpy as np
from scipy.stats import beta

# Distribution name and prefix
prefix = "1D_beta50020"
description = "Beta(500, 20)"

a = 500
b = 20

# Function to generate X values
def extract_samples(size):
    return np.random.beta(a, b, size=size)

# Function to calculate y values (PDF)
def compute_pdf(X):
    return beta.pdf(np.sort(X), a, b)